src/SoundOptionsModal.test.js

```diff
@@ -0,0 +1,154 @@
+import messages from './messages.json';
+import SoundOptionsModal from './SoundOptionsModal';
+import ToggleSwitch from './ToggleSwitch';
+
+configure({ adapter: new Adapter() });
+
+const defaultSoundOptionsModalProps = {
+    audioEnabled: true,
+    announcementsEnabled: true,
+    sonificationEnabled: true,
+    show: true
+};
+
+function createMountSoundOptionsModal(props) {
+    const mockOnCancel = jest.fn();
+    const mockOnSave = jest.fn();
+    const wrapper = mount(
+        React.createElement(
+            SoundOptionsModal,
+            Object.assign(
+                {},
+                defaultSoundOptionsModalProps,
+                {
+                    onCancel: mockOnCancel,
+                    onChangeSoundOptions: mockOnSave
+                },
+                props
+            )
+        ),
+        {
+            wrappingComponent: IntlProvider,
+            wrappingComponentProps: {
+                locale: 'en',
+                defaultLocale: 'en',
+                messages: messages.en
+            }
+        }
+    );
+
+    return {
+        wrapper,
+        mockOnCancel,
+        mockOnSave
+    };
+}
+
+function getAudioToggleSwitches(wrapper) {
+    return wrapper.find(ToggleSwitch);
+}
+
+function getCancelButton(wrapper) {
+    return wrapper.find('.TextButton--secondaryButton');
+}
+
+function getSaveButton(wrapper) {
+    return wrapper.find('.TextButton--primaryButton');
+}
+
+function getWrapperState(wrapper) {
+    return wrapper.children().at(0).instance().state;
+}
+
+describe('When rendering audio toggles', () => {
+    test('Three toggle switches should render in following order: "All Sounds", "Musical Sounds" and "Audio Announcements"', () => {
+        expect.assertions(8);
+        const { wrapper } = createMountSoundOptionsModal();
+        // No options should have disabled class when allSound toggle is on
+        expect(wrapper.find('.SoundOptions__option--disabled').length).toBe(0);
+        const audioToggleSwitches = getAudioToggleSwitches(wrapper);
+        expect(audioToggleSwitches.length).toBe(3);
+        expect(audioToggleSwitches.get(0).props.id).toBe('sound-options-allsounds');
+        expect(audioToggleSwitches.get(0).props.value).toBe(true);
+        expect(audioToggleSwitches.get(1).props.id).toBe('sound-options-musicalSounds');
+        expect(audioToggleSwitches.get(1).props.value).toBe(true);
+        expect(audioToggleSwitches.get(2).props.id).toBe('sound-options-announcements');
+        expect(audioToggleSwitches.get(2).props.value).toBe(true);
+    });
+    test('Changing audioEnabled, announcementsEnabled and sonificationEnabled properties to false should turn off corresponding toggle siwtches', () => {
+        expect.assertions(8);
+        const { wrapper } = createMountSoundOptionsModal({ audioEnabled: false, announcementsEnabled: false, sonificationEnabled: false });
+        // Other options than allSounds should have disabled class when audioEnabled is false
```

Comment:
The assertion below is not verifying what this comment states. Only that there are 2 instances of `.SoundOptions__option--disabled`. You don't check where they are.

Reply:
ToggleSwitch for "All Sounds" never becomes disabled, and that's why there are 2 instances of .SoundOptions__option--disabled. The comment is saying that when "All Sounds" is toggled off, other two switches will have disabled class, so there are 2 instances. Should I update the test case to be more specific about which toggle become disabled?

Reply:
Yes, please. Keep the existing assertion about the number of instances of the `.SoundOptions__option--disabled` class. But then also add checks that they are in the expected places.